libsolutil/Numeric.h

```diff
@@ -33,6 +33,9 @@
 #if defined(__GNUC__) && !defined(__clang__) && (__GNUC__ >= 12)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#pragma GCC diagnostic ignored "-Warray-bounds"
+#pragma GCC diagnostic ignored "-Wstringop-overread"
+#pragma GCC diagnostic ignored "-Waggressive-loop-optimizations"
 #endif
```

Comment:
At some point we should really figure out why we need this at all - if the build was set up properly, it should ignore issues inside system headers in general and we shouldn't have to disable these here in particular... might just be that it's hard for gcc to tell where the error is actually coming from, since all of these happen in template instantiations...

Anyways, I'd just keep it like this for now.

Reply:
But if it's the templates, GCC 11 should have had the same problem...